### PR TITLE
【paddle.distributed.fleet】add data_generator in distributed.fleet.dataset

### PR DESCRIPTION
**Pull request by yaoxuefeng6**
### PR types

Others
### PR changes

Others
### Describe

add data_generator in paddle.distributed.fleet.dataset to illustrate dataset class well
Based on comments of this pr:https://github.com/PaddlePaddle/Paddle/pull/27133#pullrequestreview-489234912
1, mark static only in new dataset api doc
2, mark deprecated in dataset api in fluid
3, example codes fit with 2.0 apis

my_data_generator class base on class MultiSlotDataGenerator 
my_data_generator.py: 
```python
import paddle
import paddle.distributed.fleet as fleet

class MyDataGenerator(fleet.MultiSlotDataGenerator):
    def generate_sample(self, line):
        def data_iter():
            for i in range(10000):
                yield ("words", [1, 2, 3, 4]), ("label", [0])

        return data_iter
if __name__ == "__main__":
    d = MyDataGenerator()
    d.run_from_stdin()
```
paddle.distributed.InMemoryDataset/QueueDataset using data_generator demo
```python
import paddle
paddle.enable_static()

slots = ["slot1", "slot2", "slot3", "slot4"]
slots_vars = []
for slot in slots:
var = paddle.static.data(name=slot, shape=[1], dtype="int64", lod_level=1)
slots_vars.append(var)

# create dataset instance directly with distributed.InMemoryDataset
dataset = paddle.distributed.InMemoryDataset()
# call init() to initialize single node related settings once.
# use my_data_generator in pipecommand
dataset.init(
    batch_size=32,
    thread_num=3,
    pipe_command="python my_data_generator.py",
    use_var=slots_vars)
```

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/simnet_dataset_reader.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 import random
 
 import paddle
-import paddle.fluid.incubate.data_generator as data_generator
+from paddle.distributed.fleet.dataset import data_generator as data_generator
```
Comment (guru4elephant) [inline python/paddle/fluid/tests/unittests/simnet_dataset_reader.py:25]: please move data_generator to paddle.distributed.fleet

File: python/paddle/distributed/fleet/data_generator/data_generator.py [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+
+import os
+import sys
+
+
+class DataGenerator(object):
+    """
+    DataGenerator is a general Base class for user to inherit
+    A user who wants to define his/her own python processing logic
+    with paddle.fluid.dataset should inherit this class.
```
Comment (jzhang533) [inline python/paddle/distributed/fleet/data_generator/data_generator.py]: 这里还是paddle.fluid.dataset

**Review by guru4elephant — APPROVED** [review]
LGTM

**Review by guru4elephant — DISMISSED** [review]
LGTM

**Review by luotao1 — APPROVED** [review]
```
2020-09-25 11:48:34 ****************
2020-09-25 11:48:34 0. You must have one RD (XiaoguangHu01,Xreki,luotao1) approval for python/paddle/distributed/__init, which manages the underlying code for fluid.
```
这个文件应该加上 @guru4elephant

**Review by jzhang533 — APPROVED** [review]
lgtm

**Pull request merged**